package/MDAnalysis/topology/RDKitParser.py

```diff
@@ -0,0 +1,249 @@
+
+
+Example
+-------
+TODO
+
+Classes
+-------
+
+.. autoclass:: RDKitParser
+   :members:
+   :inherited-members:
+
+"""
+from __future__ import absolute_import, division
+
+import logging
+import numpy as np
+
+from .base import TopologyReaderBase, change_squash
+from . import guessers
+from ..core.topologyattrs import (
+    Atomids,
+    Atomnames,
+    Atomtypes,
+    Elements,
+    Masses,
+    Charges,
+    Bonds,
+    Resids,
+    Resnums,
+    Resnames,
+    Segids,
+    AltLocs,
+    ChainIDs,
+    Occupancies,
+    Tempfactors,
+)
+from ..core.topology import Topology
+
+logger = logging.getLogger("MDAnalysis.topology.RDKitParser")
+
+
+class RDKitParser(TopologyReaderBase):
+    """
+    For RDKit structures
+    """
+    format = 'RDKIT'
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this Parser read object *thing*?
+
+        .. versionadded:: 1.0.0
+        """
+        try:
+            from rdkit import Chem
+        except ImportError:  # if no rdkit, probably not rdkit
+            return False
+        else:
+            return isinstance(thing, Chem.Mol)
+
+    def parse(self, **kwargs):
+        """Parse RDKit into Topology
+
+        Returns
+        -------
+        MDAnalysis *Topology* object
+        """
+
+        mol = self.filename
+
+        # Atoms
+        names = []
+        resnums = []
+        resnames = []
+        elements = []
+        masses = []
+        charges = []
+        ids = []
+        atomtypes = []
+        segids = []
+        altlocs = []
+        chainids = []
+        occupancies = []
+        tempfactors = []
+        for atom in mol.GetAtoms():
+            ids.append(atom.GetIdx())
+            elements.append(atom.GetSymbol())
+            masses.append(atom.GetMass())
+            mi = atom.GetMonomerInfo()
+            if mi: # atom name and residue info are present
+                names.append(mi.GetName().strip())
+                resnums.append(mi.GetResidueNumber())
+                resnames.append(mi.GetResidueName())
+                segids.append(mi.GetSegmentNumber())
+                altlocs.append(mi.GetAltLoc().strip())
+                chainids.append(mi.GetChainId().strip())
+                occupancies.append(mi.GetOccupancy())
+                tempfactors.append(mi.GetTempFactor())
+            else:
+                # atom name (MOL2 only)
+                try:
+                    names.append(atom.GetProp('_TriposAtomName'))
+                except KeyError:
+                    pass
+                # atom type (MOL2 only)
+                try:
+                    atomtypes.append(atom.GetProp('_TriposAtomType'))
+                except KeyError:
+                    pass
+                # gasteiger charge (computed):
+                # if the user took the time to compute them, make it a priority
+                # over charges read from a MOL2 file
+                try:
+                    charges.append(atom.GetProp('_GasteigerCharge'))
```

Comment:
We'd probably want to warn users here, if they happen to have both types of charges available, that we are defaulting to Gasteiger over the user-read ones.